src/auth/auth.controller.ts

```diff
@@ -12,7 +12,7 @@ import { AuthGuard } from '@nestjs/passport';
 import { Request } from 'express';
 
 import { AuthService } from './auth.service';
-import { RegisterInput } from './dto/register.dto';
+import { RegisterInput, RegisterOutput } from './dto/register.dto';
```

Comment:
I think we were discussing a bit in our previous meeting how to do naming convention. 
@war1oc @asad-mlbd @Saad-Amjad 

How about in this case:
1. CreateUserRequest / CreateUserResponse
2. CreateUserInput / CreateUserOutput
3. RegisterInput / RegisterOutput
4. CreateUserIncoming / CreateUserOutgoing

My personal preference is to use `CRUD` terminologies over these ones as well.
`Add` -> `Create`
`Remove` -> `Delete`

This discussion is not a blocker, but I just want to hear everyone's opinion once before we make this the default. :bow:

Reply:
Yes, I also preferred strict names for basic crud of resources.

GET index (for list)
GET show (for one item) 
POST store
PATCH update
DELETE destroy

Reply:
Another small discussion point, let's start the versioning of url routes from the start, like v1/{resource}.

Reply:
@yashmurty, I also think we should definitely fix a naming convention that we want to follow.
What I have followed here is function name + input/output. Also please note that auth module and user module both are involved when a registration of a user is going on, so both modules have their own DTOs. One is concerned with registration and the other creating a user in the system.  
`CreateUser` can be used as the domain functionality in user module and hence `CreateUserInput` or `-Request` as the DTO.  
`Register` or `RegisterNewUser` can be the domain functionality in auth module and hence `RegisterInput` etc as the DTO.  
I'm fine with either: `Input/Request`.

Reply:
@war1oc I absolutely agree with your explanation on this one! 
> Also please note that auth module and user module both are involved when a registration of a user is going on, so both modules have their own DTOs. One is concerned with registration and the other creating a user in the system.

I think `Input/Output` might be a more logical choice over `Request/Response`.
But what about `Add/Remove` vs `Create/Delete`?

Reply:
> Another small discussion point, let's start the versioning of url routes from the start, like v1/{resource}.

@Saad-Amjad Sure. Let's bring this up in the next NodeJS Sync Meeting. :smile:

Reply:
Here are my thoughts (without logic, mostly what feels good in my eyes)
- I prefer `UserCreateInput` over `CreateUserInput`
- `UserCreateRequest` sounds more sexy than `UserCreateInput`
- One technical problem naming with postfix `Request` or `Response` is It is not actually the request object. It is actually the request body. Like the request means body, header, cookies, and so many other things (am I thinking too much!)
- One concern, we are planning to have a dedicated DTO class for each controller method. But cases where we have to return the same object from 2 different controller methods (like `/me`,  `user/1`) we have to keep 2 class. Which does not look good to me.

Reply:
@asad-mlbd 
> One concern, we are planning to have a dedicated DTO class for each controller method. 

I didn't know we had this in mind. I feel it's completely okay to use the same DTO class for multiple controller methods. 
Is there a reason why we may wanna have a dedicated declaration for each controller method?

Reply:
> > One concern, we are planning to have a dedicated DTO class for each controller method.
> 
> I didn't know we had this in mind. I feel it's completely okay to use the same DTO class for multiple controller methods.
> Is there a reason why we may wanna have a dedicated declaration for each controller method?

We didn't had any discussion like that, but if we name response DTO as CreateUserResponse, that means we will have different Response DTO for each controller method, isn't it ?

I think for the response, we can give logical names rather than name based on the controller method.
Like `UserResponse`,  `UserListResponse` , etc

Reply:
@asad-mlbd you are spot on! I missed thinking that far. 
> I think for the response, we can give logical names rather than name based on the controller method.
Like UserResponse, UserListResponse , etc

I think this is a great suggestion and we could do this.